### PR TITLE
Digital post

### PR DESCRIPTION
**Pull request by rimi-itk**
https://leantime.itkdev.dk/dashboard/home#/tickets/showTicket/1015

Optimizes handling of digital post to make sure that we don't send a superfluously big payload to SF1601's KombiPostAfsend endpoint.

nsek dmzwebstgitk03 itkdev dk_case_39310cda-4e2c-4a85-b59e-d7d7fdc0bcbe_hearing_f760b679-a896-4776-a6e0-6f79d9f6a7ed_request_create (1)
nsek dmzwebstgitk03 itkdev dk_case_39310cda-4e2c-4a85-b59e-d7d7fdc0bcbe_hearing_f760b679-a896-4776-a6e0-6f79d9f6a7ed_request_create

Also contains minor tweaks and fixes to make developers and code analysis tools happy.

### CODE REVIEW

File: assets/app.js [hunk 1]
```diff
@@ -9,4 +9,24 @@ const tooltipTriggerList = document.querySelectorAll('[data-bs-toggle="tooltip"]
 window.addEventListener('load', () => {
   [...tooltipTriggerList].map(tooltipTriggerEl => new bootstrap.Tooltip(tooltipTriggerEl))
   window.dispatchEvent(new Event('ajaxload'))
+
+  const recipientSelector = '.digital-post-recipient'
+  const digitalPostRecipientWrapper = document.querySelector(recipientSelector)?.parentNode?.parentNode
+  const digitalPostAttachmentsInfo = document.querySelector('.digital-post-attachments-info')
+  if (digitalPostRecipientWrapper && digitalPostAttachmentsInfo) {
+    const recipients = digitalPostRecipientWrapper.querySelectorAll(recipientSelector)
+    const updateStuff = () => {
```
Comment (jekuaitk) [inline assets/app.js:18]: Consider renaming although it does indeed update stuff

File: assets/app.js [hunk 2]
```diff
@@ -9,4 +9,24 @@ const tooltipTriggerList = document.querySelectorAll('[data-bs-toggle="tooltip"]
...
+      const recipientTheCannotReceiveDigitalPost = [...recipients]
```
Comment (jekuaitk) [inline assets/app.js:19]:
```suggestion
      const recipientsThatCannotReceiveDigitalPost = [...recipients]
```

File: assets/app.js [hunk 3]
```diff
@@ -9,4 +9,24 @@ const tooltipTriggerList = document.querySelectorAll('[data-bs-toggle="tooltip"]
...
+        .filter(el => !el.dataset.digitalPostAllowed &&
+                el.querySelector('input[type="checkbox"]').checked);
+
+      [...digitalPostAttachmentsInfo.querySelectorAll('.digital-post-attachments-info-item')]
+        .forEach(el => {
+          el.hidden = !(el.dataset.digitalPostAllowed ^ (recipientTheCannotReceiveDigitalPost.length > 0))
```
Comment (jekuaitk) [inline assets/app.js:25]: If i understand correctly this handles which div should be hidden and which should not in `digital_post_attachments_info.html.twig`, but it took me a while to get my head around.

File: assets/content/content.scss [hunk 4]
```diff
@@ -3,3 +3,23 @@ th, td {
...
+  &[data-digital-post-allowed=""] {
+    .digital-post-message {
+      color: red;
+    }
+  }
+}
+
+// .tvist1-can-receive-digital-post {
+//   color: green;
+// }
```
Comment (jekuaitk) [inline assets/content/content.scss:18]:
```suggestion

```

File: src/Service/SF1601/DigitalPoster.php [hunk 5]
```diff
@@ -67,17 +92,26 @@ public function sendDigitalPost(DigitalPost $digitalPost, DigitalPost\Recipient
...
+            // will require us to duplicate the content in the call to "kombiPostAfsend".
+            $meMoMessage = null;
+            $forsendelse = null;
+            $type = $this->canReceive(SF1601::FORESPOERG_TYPE_DIGITAL_POST, $recipient->getIdentifier() ?? '') ? SF1601::TYPE_DIGITAL_POST : SF1601::TYPE_FYSISK_POST;
 
-            $forsendelseOptions = [
-                ForsendelseHelper::FORSENDELSES_TYPE_IDENTIFIKATOR => $this->options['sf1601']['forsendelses_type_identifikator'],
-            ];
-            $forsendelse = $this->forsendelseHelper->createForsendelse($digitalPost, $recipient, $forsendelseOptions);
+            if (in_array($type, [SF1601::TYPE_DIGITAL_POST, SF1601::TYPE_AUTOMATISK_VALG], true)) {
```
Comment (jekuaitk) [inline src/Service/SF1601/DigitalPoster.php:101]: From what i gather `$type` can enver be `SF1601::TYPE_AUTOMATISK_VALG` here.

File: src/Service/SF1601/DigitalPoster.php [hunk 6]
```diff
@@ -67,17 +92,26 @@ public function sendDigitalPost(DigitalPost $digitalPost, DigitalPost\Recipient
...
+                $meMoMessage = $this->meMoHelper->createMeMoMessage($digitalPost, $recipient, $meMoOptions);
+            }
+
+            if (in_array($type, [SF1601::TYPE_FYSISK_POST, SF1601::TYPE_AUTOMATISK_VALG], true)) {
```
Comment (jekuaitk) [inline src/Service/SF1601/DigitalPoster.php:105]: From what i gather `$type` can enver be `SF1601::TYPE_AUTOMATISK_VALG` here.

**Review by jekuaitk — CHANGES_REQUESTED** [review]

File: src/Service/SF1601/DigitalPoster.php (hunk 5 above)
Comment (rimi-itk) [reply src/Service/SF1601/DigitalPoster.php:101]: You're absolutely right, but there's nothing wrong with future-proofing a solution to a problem.

**Review by jekuaitk — APPROVED** [review]

**Pull request merged**